src/node_http2.cc

```diff
@@ -2284,7 +2284,7 @@ ssize_t Http2Stream::Provider::Stream::OnRead(nghttp2_session* handle,
     return NGHTTP2_ERR_DEFERRED;
   }
 
-  if (stream->queue_.empty() && !stream->IsWritable()) {
+  if (stream->available_outbound_length_ == 0 && !stream->IsWritable()) {
```

Comment:
why this change? Do we need a test for this?

Reply:
Right now, Node.JS sends:

 - `DATA` frame with payload (as many as needed)
 - `DATA` frame (EOF flag) with empty payload

This empty frame is unnecessary, we can set the EOF flag on the last and be done.
This is what this change does. If you look at the lines above:

https://github.com/nodejs/node/blob/14a0cd8cb368b1e7f781b883d889e62ee841a92f/src/node_http2.cc#L2264-L2272

We try to include any remaining stream data, and decrement `available_outbound_length_`.
So, it's sufficient to test for `stream->available_outbound_length_ == 0` (i.e. all data has been sent *or* is being sent in this frame).

Reply:
As for tests: this is a protocol-level optimization, so I don't see the need for tests (and I wouldn't know how to test for this, anyway).

Reply:
@mcollina PTAL